addons/storyshots/src/index.js

```diff
@@ -31,11 +31,14 @@ export default function testStorySnapshots(options = {}) {
   if (isStorybook) {
     storybook = require.requireActual('@storybook/react');
     // eslint-disable-next-line
-    const loadBabelConfig = require('@storybook/react/dist/server/babel_config').default;
+    const loadBabelConfig = require('@storybook/react/dist/server/babel_config')
+      .default;
     const configDirPath = path.resolve(options.configPath || '.storybook');
     configPath = path.join(configDirPath, 'config.js');
 
     const babelConfig = loadBabelConfig(configDirPath);
+    // We set this in the default babel config, but it is webpack-only
+    delete babelConfig.cacheDirectory;
```

Comment:
OK, done! Please take a look at the last commit.

Along the way this actually improves the behaviour in RN, before it wasn't using a cache dir if you had a custom `.babelrc` (which I reckon you *always* would in RN)